src/core/webidl.js

```diff
@@ -231,7 +232,9 @@ function getNameAndId(defn, parent = "") {
 
 function resolveNameAndId(defn, parent) {
   let name = getDefnName(defn);
-  let idlId = getIdlId(name, parent);
+  // For getters, setters, etc. "annonymous-getter",
+  const prefix = defn.special && defn.name === "" ? "annonymous-" : "";
```

Comment:
I wonder this could just be `-` so that the id would be `dom-foo--getter`. Because 1) the "named" version also has the behavior of "anonymous" one as [it's just a syntax sugar on top of it](https://heycam.github.io/webidl/#example-50d7ab88) 2) and it's just shorter 😄

(typo: a"n"onymous)

Reply:
the "--" looks weird to me... I kinda want it to be explicitly named "anonymous". Having said that, the ids are just for us so not super concerned about length.